spark/src/main/spark-3.5/org/apache/spark/sql/comet/shims/ShimCometScanExec.scala

```diff
@@ -49,16 +48,14 @@ trait ShimCometScanExec {
     filePartitions,
     readSchema,
     fileConstantMetadataColumns,
-    Map.empty,
+    fsRelation.fileFormat.fileConstantMetadataExtractors,
     options)
 
   protected def invalidBucketFile(path: String, sparkVersion: String): Throwable =
-    new SparkException("INVALID_BUCKET_FILE", Map("path" -> path), null)
+    QueryExecutionErrors.invalidBucketFile(path)
 
-  protected def isNeededForSchema(sparkSchema: StructType): Boolean = {
-    // TODO: remove after PARQUET-2161 becomes available in Parquet (tracked in SPARK-39634)
-    ShimFileFormat.findRowIndexColumnIndexInSchema(sparkSchema) >= 0
-  }
+  // see SPARK-39634
+  protected def isNeededForSchema(sparkSchema: StructType): Boolean = false
```

Comment:
We still have this in the 4.0 shim. Is this not needed any more?

Reply:
nvm. I was looking at the wrong file. This is also changed in the 4.0 shim.

Reply:
I updated this to match the version in the 4.0 shims because https://issues.apache.org/jira/browse/SPARK-39634 is marked as fixes in 3.5